pkg/runtask/v1beta1/kubernetes_test.go

```diff
@@ -0,0 +1,240 @@
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pkg/errors"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func fakeGetClientsetOk() (cs *clientset.Clientset, err error) {
+	return &clientset.Clientset{}, nil
+}
+
+func fakeGetOk(cs *clientset.Clientset, name, namespace string,
+	opts metav1.GetOptions) (*apis.RunTask, error) {
+	return &apis.RunTask{}, nil
+}
+
+func fakeGetErr(cs *clientset.Clientset, name, namespace string,
+	opts metav1.GetOptions) (*apis.RunTask, error) {
+	return &apis.RunTask{}, errors.New("some error")
+}
+
+func fakeSetClientsetOk(k *Kubeclient) {
+	k.clientset = &clientset.Clientset{}
+}
+
+func fakeSetClientsetNil(k *Kubeclient) {
+	k.clientset = nil
+}
+
+func fakeGetClientsetNil() (clientset *clientset.Clientset, err error) {
+	return nil, nil
+}
+
+func fakeGetClientsetErr() (clientset *clientset.Clientset, err error) {
+	return nil, errors.New("Some error")
+}
+
+func fakeClientSet(k *Kubeclient) {}
+
+func TestWithDefaults(t *testing.T) {
+	tests := map[string]struct {
+		getFn              getFunc
+		getClientsetFn     getClientsetFunc
+		expectGet          bool
+		expectGetClientset bool
+	}{
+		// The current implementation of WithDefaults method can be
+		// tested using these two combinations only.
+		"When mockclient is empty": {nil, nil, false, false},
+		"When mockclient contains all of them": {fakeGetOk,
+			fakeGetClientsetOk, false, false},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			fc := &Kubeclient{
+				get:          mock.getFn,
+				getClientset: mock.getClientsetFn,
+			}
+
+			fc.withDefaults()
+			if mock.expectGet && fc.get == nil {
+				t.Fatalf(`test %s failed: expected non-nil fc.get
+but got %v`, name, fc.get)
+			}
+			if mock.expectGetClientset && fc.getClientset == nil {
+				t.Fatalf(`test %s failed: expected non-nil fc.getClientset
+but got %v`, name, fc.getClientset)
+			}
+		})
+	}
+}
+
+func TestWithClientset(t *testing.T) {
+	tests := map[string]struct {
+		clientset    *clientset.Clientset
+		isKubeClient bool
+	}{
+		"Clientset is empty":     {nil, false},
+		"Clientset is not empty": {&clientset.Clientset{}, true},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			kc := KubeClient(WithClientset(mock.clientset))
+			if mock.isKubeClient && kc.clientset == nil {
+				t.Fatalf(`test %s failed, expected non-nil fake.clientset
+but got %v`, name, kc.clientset)
+			}
+			if !mock.isKubeClient && kc.clientset != nil {
+				t.Fatalf(`test %s failed, expected nil fake.clientset
+but got %v`, name, kc.clientset)
+			}
+		})
+	}
+}
+
+func TestWithNamespace(t *testing.T) {
+	tests := map[string]struct {
+		namespace    string
+		isKubeClient bool
```

Comment:
can we rename this to `expectKubeClient`

Reply:
done.